.ci/Jenkinsfile

```diff
@@ -371,9 +371,15 @@ def withCloudTestEnv(Closure body) {
   if (!aws.containsKey('secret_key')) {
     error("${AWS_ACCOUNT_SECRET} doesn't contain 'secret_key'")
   }
+  // GCP
```

Comment:
Unfortunately, it's not so simple in this case, but no worries, we'll sort it out :)

Could you please try something similar to this:

```groovy
withCredentials([string(credentialsId: "${env.JOB_GCS_BUCKET_INTERNAL}", variable: 'GOOGLE_CREDENTIALS')]) {
  maskedVars.add([var: 'GOOGLE_CREDENTIALS', password: env.GOOGLE_CREDENTIALS]);
}
```

The idea is to pull the env from credentials and place it in the masked vars. Be careful not to leak the confidential data with some "print" :)

Reply:
well, I tried :D I updated the code with your suggestion, still I do not understand who is going to set `GOOGLE_CREDENTIALS` variable in the GCP project :) (If there is any docs about this, happy to read them)

Reply:
Elastic-package uses the env.yml file to pass selected env variables to the Terraform docker container.

I see that you also use GCP_PROJECT_ID. Is it required? If so, then we need to set it here too (hardcode?)

Reply:
`GCP_PROJECT_ID` is required, and is the project where resources will be created into, and where the credentials passed are valid. Should it be a masked var as well?

Reply:
> Should it be a masked var as well? 

I don't think it makes any difference. If it's simpler to implement, keep it masked.

> Which project should I use?

`elastic-observability`, that's the one that contains an internal bucket with logs. If you need a numeric ID, log in to the GCP control panel and copy it from there.

Reply:
I noticed that you mistyped it. It should be:

```groovy
withCredentials([string(credentialsId: "${env.JOB_GCS_EXT_CREDENTIALS}", variable: 'GOOGLE_CREDENTIALS')]) {
```

Reply:
@v1v We're having a hard time figuring out why the Jenkins pipeline fails here. Apparently, there isn't any error logged. We want to read the credentials for the `elastic-observability` account and store them in GOOGLE_CREDENTIALS. 

Would you mind taking a look?

Reply:
Fixed and I added `GCP_PROJECT_ID`, but CI is still red without errors. I'm not sure why there are no errors in the CI tasks.

Reply:
A bit of clarity about debugging errors:

1) Genuine errors are reported correctly in the Blue Ocean UI
2) Pipeline errors are not reported correctly in the Blue Ocean UI

The problem is related to the credentials type:

```
11:09:05  ERROR: Credentials 'beats-ci-gcs-plugin-file-credentials' is of type 'Secret file' where 'org.jenkinsci.plugins.plaincredentials.StringCredentials' was expected

```


The below command helped to debug the error:

```
curl -s https://beats-ci.elastic.co/job/Ingest-manager/job/elastic-package/job/PR-701/5/consoleText | grep -i error
```

I'll raise an issue to be able to show errors related to `2)` since the ones for `1)` are reported in the GitHub comment, but the ones for `2)` are not (I don't know if I'll be success)

Reply:
BTW, sorry that you had some issues while debugging these errors. It's not a great user experience!

The traditional jenkins UI is the last resort to debug errors that are not easy to catch

Reply:
Thank you, @v1v for bringing it up!

What we're trying to achieve here is to extract GOOGLE_CREDENTIALS for the `elastic-observability` account and pass it to the system tests runner (elastic-package [Terraform service deployer](https://github.com/elastic/elastic-package/blob/main/docs/howto/system_testing.md#terraform-service-deployer)). We have already done with AWS credentials (access key, secret access key) and it works well.

Now, it's time to fetch credentials for GCP. The `GOOGLE_CREDENTIALS` env will be passed to the Terraform service deployer (Docker container) to execute TF scenario and create GCP resources.

PS. Probably same story for `GCP_PROJECT_ID`.